db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1234,7 +1234,10 @@ Status DBImpl::CompactRangeInternal(const CompactRangeOptions& options,
             check_overlap_within_file = false;
           }
         }
-        if (!check_overlap_within_file) {
+        if (!check_overlap_within_file ||
+            (!overlap && options.bottom_level_check_range_overlap &&
```

Comment:
Could you add a comment to elaborate on why this can ensure the SST file in the bottom level with a huge range can be force partitioned with manual compaction.

Reply:
done.